### PR TITLE
Update DataImportCron class constructor, documentation and add 'to_dict' parameters validation

### PR DESCRIPTION
**Pull request by oharan2**

### CODE REVIEW

File: ocp_resources/data_import_cron.py [hunk 1]
```diff
@@ -57,8 +57,14 @@ def to_dict(self):
         super().to_dict()
         if not self.yaml_file:
             if self.image_stream and self.url:
-                raise ValueError("imageStream and url can not coexist")
-
+                raise ValueError("imageStream and url cannot coexist")
+            if not all(
+                [self.size, self.managed_data_source, self.pull_method, self.schedule]
```
Comment (myakove) [inline ocp_resources/data_import_cron.py:62]: `self.managed_data_source` and `self.schedule` marked as required in the API docs but have defaults so we cannot force the user to send them.

Comment (myakove) [reply ocp_resources/data_import_cron.py:62]: `self.size` is part of `storage` key which is not mandatory.

Comment (oharan2) [reply ocp_resources/data_import_cron.py]: Please checkout latest commits [approach](https://github.com/RedHatQE/openshift-python-wrapper/pull/1490/commits/2b9817657389968edf1c02c4a1de09c4252c4f2b#diff-bf49cfb5d402f8bb0e46e2125739760ec738f1a91689835bff8f8995f87beef1R99).
I can revert it in a second to the last usage of creating a storage on [L70](https://github.com/RedHatQE/openshift-python-wrapper/pull/1490/files#diff-bf49cfb5d402f8bb0e46e2125739760ec738f1a91689835bff8f8995f87beef1L70) and set also when `self.size` is None.

Comment (oharan2) [reply ocp_resources/data_import_cron.py]: Example output:
```
{'source': {'registry': {'pullMethod': '..', 'imageStream': '..'}}, 
'storage': {'resources': {'requests': {'storage': None}}, 'storageClassName': '..'}}
```

**Review by omrirh — APPROVED** [review]

File: ocp_resources/data_import_cron.py [hunk 2]
```diff
@@ -57,21 +57,20 @@ def to_dict(self):
...
                     "spec": {
-                        "template": {
-                            "spec": {
-                                "source": {"registry": {"pullMethod": self.pull_method}},
-                                "storage": {"resources": {"requests": {"storage": self.size}}},
-                            }
-                        }
+                        "template": {"spec": {"source": {"registry": {"pullMethod": self.pull_method}}}},
+                        "managedDataSource": self.managed_data_source,
+                        "schedule": self.schedule,
```
Comment (rnetser) [inline ocp_resources/data_import_cron.py:68]: add only if provided

File: ocp_resources/data_import_cron.py [hunk 3]
```diff
@@ -57,21 +57,20 @@ def to_dict(self):
...
                 {
                     "spec": {
-                        "template": {
-                            "spec": {
-                                "source": {"registry": {"pullMethod": self.pull_method}},
-                                "storage": {"resources": {"requests": {"storage": self.size}}},
-                            }
-                        }
+                        "template": {"spec": {"source": {"registry": {"pullMethod": self.pull_method}}}},
+                        "managedDataSource": self.managed_data_source,
```
Comment (rnetser) [inline ocp_resources/data_import_cron.py:67]: add only if provided

**Review by rnetser — CHANGES_REQUESTED** [review]

**Review by rnetser — CHANGES_REQUESTED** [review]
Comments were not addressed

**Review by rnetser — APPROVED** [review]

**Review by omrirh — APPROVED** [review]

**Review by myakove — APPROVED** [review]

**Pull request merged**